tests/testthat/test-module_teal.R

```diff
@@ -1,97 +1,1843 @@
+        )
+      ),
+      expr = {
+        session$setInputs(`teal_modules-active_tab` = "module_1")
+        testthat::expect_s3_class(modules_output$module_1(), "FilterPanelAPI")
+      }
+    )
+  })
+
+  testthat::it("srv_teal_module.teal_module passes Reporter if specified", {
+    shiny::testServer(
+      app = srv_teal,
+      args = list(
+        id = "test",
+        data = teal.data::teal_data(iris = iris, mtcars = mtcars),
+        modules = modules(
+          module("module_1", server = function(id, reporter) reporter)
+        )
+      ),
+      expr = {
+        session$setInputs(`teal_modules-active_tab` = "module_1")
+        testthat::expect_s3_class(modules_output$module_1(), "Reporter")
+      }
+    )
+  })
 })
 
-testthat::test_that("srv_teal when teal_data_rv changes, datasets_reactive is initialized as list of FilteredData", {
-  data <- teal.data::teal_data(iris1 = iris, mtcars1 = mtcars)
-  shiny::testServer(
-    app = srv_teal,
-    args = list(
-      id = "test",
-      teal_data_rv = reactiveVal(NULL),
-      modules = modules(
-        example_module(label = "iris_tab"),
-        example_module(label = "mtcars_tab")
-      )
-    ),
-    expr = {
-      teal_data_rv(data)
-      checkmate::expect_list(datasets_reactive(), types = "FilteredData")
-    }
-  )
+testthat::describe("srv_teal filters", {
+  testthat::describe("slicesGlobal", {
+    testthat::it("is set to initial filters when !module_specific", {
+      init_filter <- teal_slices(
+        teal_slice("iris", "Species"),
+        teal_slice("mtcars", "cyl"),
+        mapping = list(
+          global_filters = c("iris Species", "mtcars cyl")
+        ),
+        module_specific = FALSE
+      )
+      shiny::testServer(
+        app = srv_teal,
+        args = list(
+          id = "test",
+          data = teal.data::teal_data(iris = iris, mtcars = mtcars),
+          modules = modules(example_module(label = "module-1"), example_module(label = "module-2")),
+          filter = init_filter
+        ),
+        expr = {
+          testthat::expect_identical(slices_global$all_slices(), init_filter)
+        }
+      )
+    })
+    testthat::it("is set to initial filters with resolved attr(, 'mapping')$<modules label> when `module_specific`", {
+      init_filter <- teal_slices(
+        teal_slice("iris", "Species"),
+        teal_slice("mtcars", "cyl"),
+        module_specific = TRUE,
+        mapping = list(
+          global_filters = c("iris Species", "mtcars cyl")
+        )
+      )
+      shiny::testServer(
+        app = srv_teal,
+        args = list(
+          id = "test",
+          data = teal.data::teal_data(iris = iris, mtcars = mtcars),
+          modules = modules(example_module(label = "module-1"), example_module(label = "module-2")),
+          filter = init_filter
+        ),
+        expr = {
+          testthat::expect_length(setdiff_teal_slices(slices_global$all_slices(), init_filter), 0)
```

Comment:
@gogonzo this is because `testthat::describe` does not see utils loaded in `zzz.R`. Weird
```suggestion
          setdiff_teal_slices <- getFromNamespace("setdiff_teal_slices", "teal.slice")
          testthat::expect_length(setdiff_teal_slices(slices_global$all_slices(), init_filter), 0)
```

Reply:
This happens only during R CMD CHECK but not during tests

```
── Error ('test-module_teal.R:732:11'): is set to initial filters with resolved attr(, 'mapping')$<modules label> when `module_specific` ──
Error in `setdiff_teal_slices(slices_global$all_slices(), init_filter)`: could not find function "setdiff_teal_slices"
```

Reply:
You can see this fails in this test in GitHub Action https://github.com/insightsengineering/teal/actions/runs/10143639530/job/28045441236#step:41:173